pkgs/development/tools/continuous-integration/github-runner/default.nix

```
@@ -276,6 +273,34 @@ stdenv.mkDerivation rec {
     wrap config.sh --prefix PATH : ${lib.makeBinPath [ glibc.bin ]}
   '';
 
+  # Script to create deps.nix file for dotnet dependencies. Run it with
+  # $(nix-build -A github-runner.passthru.createDepsFile)/bin/create-deps-file
+  # and copy result file into github-runner folder.
+  # Inspired by passthru.fetch-deps in pkgs/build-support/build-dotnet-module/default.nix
+  passthru.createDepsFile = writeShellApplication {
+    name = "create-deps-file";
+    runtimeInputs = [ dotnetSdk nuget-to-nix ];
+    text = ''
+      printf "\n* Setup workdir\n"
+      workdir="$(mktemp -d /tmp/github-runner.XXX)"
```

Just for consistency:

```suggestion
      workdir="$(mktemp -d /tmp/${pname}.XXX)"
```